### PR TITLE
V/0.5.0/tjr

### PR DESCRIPTION
**Pull request by troyraen**
Updates in this PR:
- Auto-scheduler: Adds auto-scheduler and related cue-response checker. Updates related docs
- Docs: Reconfigures ReadtheDocs so it works again (includes reorganizing the files). Adds tutorials.
- stream-looper VM and related ztf-loop Pub/Sub topic are setup in version_tracking/v0.5.0/stream-looper.md

### CODE REVIEW

File: broker/broker_utils/README.md [hunk 1]
```diff
@@ -0,0 +1,9 @@
+# `pgb_broker_utils`
+
+- `pip install pgb_broker_utils`
+- `import broker_utils`
+
+Note that the schema maps are stored in a Cloud Storage bucket and are not packaged with this module.
```
Comment (wmwv) [inline broker/broker_utils/README.md:6]: What implications does this have for future reproducibility?

File: broker/broker_utils/broker_utils/beam_transforms.py [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+                    psource.pop(co, None)
+
+        elif schema_map['SURVEY'] == 'ztf':
+            alertStripped = {k:v for k, v in alertDict.items() if k not in cutouts}
+
+        return [alertStripped]
+
+
+class ExtractDIASource(DoFn):
+    """Extract the DIA source` fields and information needed for provinance
```
Comment (wmwv) [inline broker/broker_utils/broker_utils/beam_transforms.py:55]: provenance

File: broker/broker_utils/broker_utils/beam_transforms.py [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+
+        return source
+
+    def process_decat(self, alertDict):
+        # get source
+        dup_cols = ['ra','dec']  # names duplicated in object and source levels
+        sourcename = lambda x: x if x not in dup_cols else f'source_{x}'
+        src = {sourcename(k):v for k,v in alertDict['triggersource'].items()}
+
+        # get info for provinance
```
Comment (wmwv) [inline broker/broker_utils/broker_utils/beam_transforms.py:77]: provenance

File: broker/broker_utils/broker_utils/beam_transforms.py [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+        # package it up and return
+        source = {**metadict, **src, 'sources_sourceids': prv_sources}
+        return [source]
+
+    def process_ztf(self, alertDict):
+        # get candidate
+        dup_cols = ['candid']  # candid is repeated, drop the one nested here
+        cand = {k:v for k,v in alertDict['candidate'].items() if k not in dup_cols}
+
+        # get info for provinance
```
Comment (wmwv) [inline broker/broker_utils/broker_utils/beam_transforms.py:97]: provenance

File: broker/cloud_functions/check_cue_response/README.md [hunk 5]
```diff
@@ -0,0 +1,24 @@
+# Check cue response
+
+This Cloud Function checks whether the broker responded appropriately to the auto-scheduler's cue. If it has not, "Critical" errors are raised which trigger a GCP alerting policy.
```
Comment (wmwv) [inline broker/cloud_functions/check_cue_response/README.md:3]: Is there one Critical error raised or multiple?

File: broker/broker_utils/requirements.txt [hunk 6]
```diff
@@ -0,0 +1,11 @@
+apache-beam[gcp]==2.25.0
```
Comment (wmwv) [inline broker/broker_utils/requirements.txt:1]: Document pinned version

File: broker/night_conductor/requirements.txt [hunk 7]
```diff
@@ -2,13 +2,16 @@ apache-beam[gcp]==2.25.0
 argparse
 astropy==3.2.1
 fastavro==0.22.3
+google-cloud-bigquery
 google-cloud-core>==1.4.1
 google-cloud-dataflow
 google-cloud-datastore>==1.15
-google-cloud-storage==1.16.1
+google-cloud-storage==1.38.0
```
Comment (wmwv) [inline broker/night_conductor/requirements.txt:9]: Document pinned version

**Review by wmwv — APPROVED** [review]
This looks good.  I haven't actually run the tutorials.

Consider adding "Learning Objectives" for the tutorials.  Clearly outline what the person should expect to learn from each tutorial.  It's hard, because they will learn many things in each tutorial, and part of the use of documentation is in searching it so you want to include all sorts of things.  But try to bring out the 3-5 main things for each tutorial.

File: broker/broker_utils/README.md (hunk 1 above)
Comment (troyraen) [reply broker/broker_utils/README.md:6]: Good question that I hadn't thought specifically about yet... The schema map files are in the code repo so, for example, a tagged release will include the schema maps used in that version. Of course this begs the question of how we will facilitate reproducibility in general which I will continue to think about. I anticipate putting something in writing for the paper.

File: broker/broker_utils/broker_utils/beam_transforms.py (hunk 2 above)
Comment (troyraen) [reply broker/broker_utils/broker_utils/beam_transforms.py:55]: thanks

File: broker/cloud_functions/check_cue_response/README.md (hunk 5 above)
Comment (troyraen) [reply broker/cloud_functions/check_cue_response/README.md:3]: One critical error per component checked (vm, dataflow job). I'll clarify in the text.

File: broker/broker_utils/requirements.txt (hunk 6 above)
Comment (troyraen) [reply broker/broker_utils/requirements.txt:1]: For posterity, referring to my comment [here](https://github.com/mwvgroup/Pitt-Google-Broker/pull/60#issuecomment-890220265). I will continue to work on these versioning issues (cleanup + document).

**Pull request merged**